src/hasher512.cpp

```diff
@@ -160,9 +160,9 @@ namespace libtorrent
 #endif
 		}
 #elif defined TORRENT_USE_LIBCRYPTO
-		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), data.size());
+		SHA512_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), std::uint32_t(data.size()));
```

Comment:
well, I was concerned only about the call to openssl ones :)
I don't know if changing the signature for the internal libtorrent sha* is good or not